x/wasm-storage/keeper/keeper.go

```diff
@@ -35,139 +78,164 @@ func (k Keeper) GetAuthority() string {
-		k.cdc.MustUnmarshal(iterator.Value(), &wasm)
+	for ; iter.Valid(); iter.Next() {
+		kv, err := iter.KeyValue()
+		if err != nil {
+			return err
+		}
 
-		if callback(wasm) {
+		if callback(kv.Value) {
 			break
 		}
 	}
+	return nil
 }
 
 // ListDataRequestWasms returns hashes and types of all Data Request Wasms
 // in the store.
 func (k Keeper) ListDataRequestWasms(ctx sdk.Context) []string {
 	var hashTypePairs []string
-	k.IterateAllDataRequestWasms(ctx, func(w types.Wasm) bool {
+	err := k.IterateAllDataRequestWasms(ctx, func(w types.Wasm) bool {
 		hashTypePairs = append(hashTypePairs, hex.EncodeToString(w.Hash)+","+w.WasmType.String())
 		return false
 	})
+	if err != nil {
+		return nil
+	}
 	return hashTypePairs
 }
 
 // ListOverlayWasms returns hashes and types of all Overlay Wasms in the store.
 func (k Keeper) ListOverlayWasms(ctx sdk.Context) []string {
 	var hashTypePairs []string
-	k.IterateAllOverlayWasms(ctx, func(w types.Wasm) bool {
+	err := k.IterateAllOverlayWasms(ctx, func(w types.Wasm) bool {
 		hashTypePairs = append(hashTypePairs, hex.EncodeToString(w.Hash)+","+w.WasmType.String())
 		return false
 	})
+	if err != nil {
+		return nil
+	}
 	return hashTypePairs
 }
 
 func (k Keeper) GetAllWasms(ctx sdk.Context) []types.Wasm {
 	var wasms []types.Wasm
-	k.IterateAllDataRequestWasms(ctx, func(wasm types.Wasm) bool {
+	err := k.IterateAllDataRequestWasms(ctx, func(wasm types.Wasm) bool {
 		wasms = append(wasms, wasm)
 		return false
 	})
-	k.IterateAllOverlayWasms(ctx, func(wasm types.Wasm) bool {
+	if err != nil {
+		return nil
+	}
+	err = k.IterateAllOverlayWasms(ctx, func(wasm types.Wasm) bool {
 		wasms = append(wasms, wasm)
 		return false
 	})
+	if err != nil {
+		return nil
+	}
 	return wasms
 }
 
+// SetProxyContractRegistry stores Proxy Contract address.
+func (k Keeper) SetProxyContractRegistry(ctx sdk.Context, address sdk.AccAddress) error {
+	return k.ProxyContractRegistry.Set(ctx, address)
+}
+
+// GetProxyContractRegistry returns Proxy Contract address.
+func (k Keeper) GetProxyContractRegistry(ctx sdk.Context) (sdk.AccAddress, error) {
+	return k.ProxyContractRegistry.Get(ctx)
+}
+
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
 // GetParams returns all the parameters for the module.
-func (k Keeper) GetParams(ctx sdk.Context) types.Params {
-	store := ctx.KVStore(k.storeKey)
-	bz := store.Get([]byte(types.KeyParams))
-
-	var params types.Params
-	k.cdc.MustUnmarshal(bz, &params)
-	return params
+func (k Keeper) GetParams(ctx sdk.Context) (types.Params, error) {
+	return k.Params.Get(ctx)
 }
 
 // SetParams sets the parameters in the store.
 func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
-	store := ctx.KVStore(k.storeKey)
-	bz, err := k.cdc.Marshal(&params)
-	if err != nil {
-		return err
-	}
-	store.Set([]byte(types.KeyParams), bz)
-
-	return nil
+	return k.Params.Set(ctx, params)
 }
```

Comment:
We don't need these methods anymore.

Reply:
Refactor: removed func [e1aaee3](https://github.com/sedaprotocol/seda-chain/pull/192/commits/e1aaee3867fdf3b44333219846a375dda0bb8c6c)